### DISCUSSION TITLE
Strange build issue

### DISCUSSION THREAD

**Discussion by danielenriquez59** [su2code/SU2#1377] [Q&A]
Hi, 

I am getting a strange build issue when I enable pywrapper. See below. 

I build my meson with: 

`python meson.py build --reconfigure -Denable-autodiff=true -Denable-directdiff=true -Denable-pywrapper=true --prefix="C:\bin"`

`WARNING: the currently checked out submodule commit in C:\Users\Danie\Documents\SU2\externals\opdi does not match the SHA-1 found in the index.
Use 'git submodule update --init C:\Users\Danie\Documents\SU2\externals\opdi' to reset the module if necessary.
SHA-1 tag stored in index does not match SHA tag stored in this script.
The Meson build system
Version: 0.54.999
Source dir: C:\Users\Danie\Documents\SU2
Build dir: C:\Users\Danie\Documents\SU2\build
Build type: native build
Project name: SU2
Project version: 7.2.0 "Blackbird"
C compiler for the host machine: gcc (gcc 8.1.0 "gcc (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0")
C linker for the host machine: gcc ld.bfd 2.30
C++ compiler for the host machine: c++ (gcc 8.1.0 "c++ (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0")
C++ linker for the host machine: c++ ld.bfd 2.30
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (C:\Users\Danie\Anaconda3\envs\vsptools\python.exe)
Did not find pkg-config by name 'pkg-config'
Found Pkg-config: NO
mpicc found: NO
Run-time dependency MPI for c found: NO (tried pkgconfig and config-tool)
mpic++ found: NO
Run-time dependency MPI for cpp found: NO (tried pkgconfig and config-tool)
Program swig found: YES (C:\Softwares\swigwin-4.0.2\swig.EXE)
Dependency python found: YES (sysconfig)
Program install.sh found: YES (sh C:\Users\Danie\Documents\SU2\SU2_PY/pySU2\install.sh)
Message: -------------------------------------------------------------------------
         |    ___ _   _ ___                                                      |
         |   / __| | | |_  )   Release 7.2.0 "Blackbird"                         |
         |   \__ \ |_| |/ /                                                      |
         |   |___/\___//___|   Meson Configuration Summary                       |
         |                                                                       |
         -------------------------------------------------------------------------

         Option          Value
         ---------------------
         TecIO:          false
         CGNS:           true
         AD (reverse):   true
         AD (forward):   true
         Python Wrapper: true
         Intel-MKL:      false
         OpenBlas:       false
         PaStiX:         false
         Mixed Float:    false
         libROM:         false

         Please be sure to add the $SU2_HOME and $SU2_RUN environment variables,
         and update your $PATH (and $PYTHONPATH if applicable) with $SU2_RUN

         Based on the input to this configuration, add these lines to your .bashrc file:

         export SU2_RUN=C:\bin/bin
         export SU2_HOME=C:\Users\Danie\Documents\SU2
         export PATH=$PATH:$SU2_RUN
         export PYTHONPATH=$PYTHONPATH:$SU2_RUN

         Use './ninja -C C:\Users\Danie\Documents\SU2\build install' to compile and install SU2

Build targets in project: 17

Found ninja.exe-1.9.0 at C:\Users\Danie\Documents\SU2\ninja.exe`

**526 files install successfully.** 

`WARNING: the currently checked out submodule commit in C:\Users\Danie\Documents\SU2\externals\opdi does not match the SHA-1 found in the index.
Use 'git submodule update --init C:\Users\Danie\Documents\SU2\externals\opdi' to reset the module if necessary.
SHA-1 tag stored in index does not match SHA tag stored in this script.
Installing SU2_CFD/src\SU2_CFD.exe to C:\bin\bin
Installing SU2_CFD/src\SU2_CFD_AD.exe to C:\bin\bin
Installing SU2_CFD/src\SU2_CFD_DIRECTDIFF.exe to C:\bin\bin
Installing SU2_DOT/src\SU2_DOT.exe to C:\bin\bin
Installing SU2_DOT/src\SU2_DOT_AD.exe to C:\bin\bin
Installing SU2_DEF/src\SU2_DEF.exe to C:\bin\bin
Installing SU2_GEO/src\SU2_GEO.exe to C:\bin\bin
Installing SU2_SOL/src\SU2_SOL.exe to C:\bin\bin
Installing SU2_PY/pySU2\_pysu2.dll to C:\bin\bin
Installing SU2_PY/pySU2\_pysu2.dll.a to C:\bin\lib
Installing SU2_PY/pySU2\_pysu2ad.dll to C:\bin\bin
Installing SU2_PY/pySU2\_pysu2ad.dll.a to C:\bin\lib
Installing C:\Users\Danie\Documents\SU2\SU2_PY\continuous_adjoint.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\compute_uncertainty.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\finite_differences.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\mesh_deformation.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\parallel_computation.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\parallel_computation_fsi.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\package_tests.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\shape_optimization.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\merge_solution.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\set_ffd_design_var.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\compute_polar.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\compute_multipoint.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\discrete_adjoint.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\direct_differentiation.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\fsi_computation.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2_CFD.py to C:\bin\bin
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/__init__.py to C:\bin\bin/SU2
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/eval/design.py to C:\bin\bin/SU2/eval
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/eval/functions.py to C:\bin\bin/SU2/eval
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/eval/gradients.py to C:\bin\bin/SU2/eval
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/eval/__init__.py to C:\bin\bin/SU2/eval
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/config.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/config_options.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/data.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/filelock.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/redirect.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/state.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/tools.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/historyMap.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/io/__init__.py to C:\bin\bin/SU2/io
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/opt/project.py to C:\bin\bin/SU2/opt
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/opt/scipy_tools.py to C:\bin\bin/SU2/opt
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/opt/__init__.py to C:\bin\bin/SU2/opt
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/adjoint.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/deform.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/direct.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/interface.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/merge.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/geometry.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/projection.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/run/__init__.py to C:\bin\bin/SU2/run
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/bunch.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/filter_adjoint.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/lhc_unif.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/misc.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/mp_eval.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/ordered_bunch.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/ordered_dict.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/plot.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/polarSweepLib.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/which.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/switch.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2/util/__init__.py to C:\bin\bin/SU2/util
Installing C:\Users\Danie\Documents\SU2\SU2_PY\FSI_tools/__init__.py to C:\bin\bin/FSI_tools
Installing C:\Users\Danie\Documents\SU2\SU2_PY\FSI_tools/FSIInterface.py to C:\bin\bin/FSI_tools
Installing C:\Users\Danie\Documents\SU2\SU2_PY\FSI_tools/FSI_config.py to C:\bin\bin/FSI_tools
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2_Nastran/__init__.py to C:\bin\bin/SU2_Nastran
Installing C:\Users\Danie\Documents\SU2\SU2_PY\SU2_Nastran/pysu2_nastran.py to C:\bin\bin/SU2_Nastran
Running custom install script 'sh C:\\Users\\Danie\\Documents\\SU2\\SU2_PY/pySU2\\install.sh C:\\Users\\Danie\\Documents\\SU2\\build\\SU2_PY/pySU2 pysu2.py'
Failed to run install script 'sh C:\\Users\\Danie\\Documents\\SU2\\SU2_PY/pySU2\\install.sh C:\\Users\\Danie\\Documents\\SU2\\build\\SU2_PY/pySU2 pysu2.py'
FAILED: meson-install
"C:\\Users\\Danie\\Anaconda3\\envs\\vsptools\\python.exe" "C:\\Users\\Danie\\Documents\\SU2\\meson.py" "install" "--no-rebuild"
ninja: build stopped: subcommand failed.`

I think it might be due to the fact that I am working in an Anaconda environment. But maybe not. Any thoughts?

**Comment by pcarruscag** [comment]
Hi,
Have you tried doing what the warning suggests?
git submodule update --init C:\Users\Danie\Documents\SU2\externals\opdi
Did you have a version before 7.2.0 in the same folder?

**Reply by danielenriquez59** [reply]
I did, and I get another error 
`error: invalid path 'include/opdi/aux/emptyMacros.hpp'
error: invalid path 'include/opdi/aux/exceptions.hpp'
error: invalid path 'include/opdi/aux/macros.hpp'
error: invalid path 'include/opdi/aux/undefineMacros.hpp'
error: invalid path 'include/opdi/tool/aux/handle.hpp'
Unable to checkout '2735b503f60163e8d64e1ac56cce46173a9fd4a9' in submodule path 'externals/opdi'`


But I didn't think the opdi files were related to the pySU2 which is where ninja fails

**Reply by pcarruscag** [reply]
That is because o the "aux" directory which cannot be used in Windows. It is fixed in develop.
The installation of the python wrapper is just a copy paste. You can try grabbing the pysu2.py file manually.

**Reply by danielenriquez59** [reply]
If I manually copy the pysu2.py file and try to 'python import' I get an error 'No module named "_pysu2"'
Is there another file/folder I should copy?
